Feature/table popover dist

reinaldocoelho opened the pull request:
#### What does this PR do?
- Changed icons of actions on table: Add row, Add column and etc.
- Included language support for table actions.
- Included translation to pt_BR.
- Inproved line functions to copy style from original element.
- Corrected bug when action is applied on table with other table inner.
#### Where should the reviewer start?
- start on the src/js/bs3/settings.js
- start on the src/js/bs3/module/Buttons.js
- start on the src/js/base/summernote-en-US.js
- start on the lang/summernote-pt-BR.js
- start on the src/js/base/editing/Table.js
#### How should this be manually tested?
- Create any table on summernote, then add, remove and change columns and rows.
#### Screenshots (if for frontend)

![image](https://cloud.githubusercontent.com/assets/889634/19368796/1fadea9c-9178-11e6-95c2-51e8b5f1bc70.png)
### Checklist
- [ ] verify icons to actions
- [ ] verify the actions with formatted tables
- [ ] verify the actions in table with other table inner

mstred committed:
Add support to custom styletag styles and formatting

mstred committed:
Add note-styletag validation in paragraph insertion

Commit:
Remove invalid `pointer-event` css property

fgallinari committed:
Update summernote-pt-BR.js

Help strings added, plus other minor changes

Commit:
Add isSupportTouch on agent.js

Commit:
Add tooltip in settings and make it automatically applied by agent.isSupportTouch

lqez committed:
Merge pull request #1914 from fgallinari/patch-1

Update summernote-pt-BR.js

lqez committed:
Revert "Update summernote-pt-BR.js"

lqez committed:
Merge pull request #1918 from summernote/revert-1914-patch-1

Revert "Update summernote-pt-BR.js"

Commit:
Refactoring for options

dany28 committed:
Adding paste event to LinkDialog

When user paste url, link text is changing but button is still disabled.
Adding paste event fixing it

Commit:
fix drop zone memory leak bug

Commit:
add semicolon

dodo360 committed:
Update README.md 

Added link to adaptation for react

hackerwins committed:
Merge pull request #1968 from Vnkitaev/patch-1

Update README.md

hackerwins committed:
move location for react-summernote

hackerwins committed:
Merge pull request #1951 from kbs0327/fix_memory_leak

fix memory leak bug in drop zone

hackerwins committed:
fixed typo.

hackerwins committed:
Merge pull request #1896 from SalesLoft/css-pointer-events

Remove invalid `pointer-event` css property

hackerwins committed:
Merge pull request #1920 from dany28/develop

Adding paste event to LinkDialog

hackerwins committed:
release 'paste' event when dialog is closed. related with $1920.

avif committed:
Fixed #1977.

Commit:
add license

Commit:
Add .nuspec for NuGet packaging

Commit:
Update .nuspec to fall back locating content files

hackerwins committed:
Merge pull request #1981 from graingert/add-license

add license

hackerwins committed:
Comment out IE9 test

hackerwins committed:
Merge branch 'develop' of https://github.com/summernote/summernote into develop

hackerwins committed:
update dist files

hackerwins committed:
Comment out Edge

parg-programador committed:
Simple translation summernote-en-BR.js

I saw the translation file and realized that the images still in English, so I decided to help.

Xiphin committed:
Update spelling errors

hackerwins committed:
Merge pull request #1998 from Xiphin/develop

Update spelling errors

hackerwins committed:
Merge pull request #1994 from parg-programador/patch-1

Simple translation summernote-en-BR.js

hackerwins committed:
Merge pull request #1983 from aliencube/develop

Add .nuspec for NuGet packaging

hackerwins committed:
Merge pull request #1978 from avif/bugfix/#1977

Fixed #1977.

hackerwins committed:
Update dist files

hackerwins committed:
Merge pull request #1976 from summernote/develop

v0.8.2

anna-git committed:
button like bold, underline, strike, italic change their visual style immediately once they get pressed

anna-git committed:
miss

anna-git committed:
added several tests for all button concerned by immediate visual change effect

anna-git committed:
#bugfix#1738 #1673

Commit:
508 compliance update on summernote.js

I used the wave tool plugin on google chrome fixed all of the errors that popped up with summernote that indicated that the button tag does not have any content within it.

Commit:
Merge pull request #1 from latifz1/508compliance_fix-1

508 compliance update on summernote.js

Commit:
Update summernote.js

Commit:
Merge pull request #4 from latifz1/feature/508_latifz1-patch-1-2

Update summernote.js

Commit:
Update summernote.js

Commit:
Merge pull request #5 from latifz1/feature/508_latifz1-patch-1-3

Update summernote.js

temuujina committed:
Mongolian translation version

Is this possible that I translate the help modal window text in this code ?

Commit:
Update summernote.js

hackerwins committed:
Merge pull request #2090 from temuujina/patch-2

Mongolian translation version

Commit:
Update summernote.js

reinaldocoelho committed:
Change table action icons and add translate

Change icons of actions in table.
Include tooltips in language files.

reinaldocoelho committed:
Copy attributes on add row/col to table

Add function to copy item attributes on add rows or cols to table.

reinaldocoelho committed:
Resolve bug on add column to table inner table.

reinaldocoelho committed:
Resolve bug on delete column logic.

When remove column in tables having table inner, the code crash table structure.

czimnick committed:
fix for angular-summernote #162

ixiumu committed:
汉化帮助

mesutcakir committed:
Math Functions And Science Characters

Math Functions And Science Characters hint avaiable.

Matematik ve özel bilimsel karakterin seçilebilmesi için gerekli örnekleme için data hazırlandı.

mesutcakir committed:
Search Problem fixed.

mesutcakir committed:
Math Functions And Science Characters hint

Math Functions And Science Characters hint avaiable.

musaffa committed:
[Fix] dont add default URL schema with onCreateLink

mstred committed:
Add support to an onDestroy event callback

pvenakis committed:
Adding the Greek translation file

abuabdul committed:
Add Tamil language support for summernote editor

abuabdul committed:
Fix jshint issues

abuabdul committed:
Update few translations in the ta-IN.js

lacivert committed:
fix statusbar shown on disableResizeEditor

fixing issue #1931

vphantom committed:
Add missing dependencies in NPM definition (#2118)

Example of what is proposed in issue #2118 to have explicit dependencies in the package.  Bootstrap itself depends simply on jquery `>=1.9.1` and I have performance issues with 3.x, so this example includes 2.x.

vphantom committed:
Base en-US wasn't updated for normal vs p

I noticed that everywhere else, last year `normal` was renamed `p`.  Seeing how the current NPM version of Summernote displays "p" instead of "Normal" in the Style drop-down, I connected the dots...

vphantom committed:
Missing French translation for 'p'

vphantom committed:
French translation was missing 'help' section

abuabdul committed:
Correct code tamil translation

onuradsay committed:
Table popover - remove table feature

add remove table button to table popover and its functionality

onuradsay committed:
Turkish translation for table popover tooltips

onuradsay committed:
Add dist files

Add dist files for Turkish translation of table popover tooltips and delete table feature

reinaldocoelho committed:
Merge pull request #1 from onuradsay/feature/table-popover-delete-table-button

Feature/table popover delete table button

abuabdul committed:
Update float and justify translation

abuabdul committed:
Add shape translation in Image

abuabdul committed:
Correct spell mistake

abuabdul committed:
Add small corrections in the translation

hackerwins committed:
Merge pull request #2228 from B2OPlus/tamil-support

Tamil Translation support

hackerwins committed:
Merge pull request #2218 from vphantom/patch-3

Missing French translation for 'style.p'

hackerwins committed:
Merge pull request #2173 from ixiumu/patch-1

汉化帮助

johnreaume commented:
Sir,
Great changes to tables!
1) you have a small bug that popup still remains while going into code view
Questions...
Are you thinking of adding split, merge column and/or row?
Why did you put this inside the summernote.js source instead of it's own plugin?

reinaldocoelho commented:
Hello @johnreaume.

I do not think of including Split or Merge columns, the added features have solved our problem.
I imagine that table editing is very basic, so I have not thought of adding the function as a plugin.

I'll check the bug, thank you.
Do you need to change the feature for the plugin?

johnreaume commented:
For the most part your version is similar to

https://www.froala.com/wysiwyg-editor/examples/full-featured

Above is not mobile friendly but adds cel/row split/merge on right click (very odd). They also provide column width resize.

I only thought a plugin would be better to keep your code outside the main source if they ever upgrade it?!?

Great work you have done...
John



Sent from my iPhone
On Mar 1, 2017, at 4:31 AM, Reinaldo Coelho Sartorelli <notifications@github.com<mailto:notifications@github.com>> wrote:


Hello @johnreaume<https://github.com/johnreaume>.

I do not think of including Split or Merge columns, the added features have solved our problem.
I imagine that table editing is very basic, so I have not thought of adding the function as a plugin.

I'll check the bug, thank you.
Do you need to change the feature for the plugin?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/xavier-d/summernote/pull/1#issuecomment-283327964>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACvxGrhWtrRStf5ODx-QM9MapHVKIXHnks5rhWUZgaJpZM4KWbsh>.

hackerwins committed:
Fixed “Can not load "PhantomJS", it is not registered!” on test.

hackerwins committed:
update libraries.

hackerwins committed:
Merge pull request #2248 from summernote/misc/update-libs

update libraries.

hackerwins committed:
Merge pull request #2239 from summernote/master

Master

hackerwins committed:
Merge pull request #2217 from vphantom/patch-2

Base en-US wasn't updated for normal vs p

hackerwins committed:
Merge pull request #2216 from vphantom/patch-1

Add missing dependencies in NPM definition (#2118)

hackerwins committed:
uncomment tests with IE9 and Edge.

hackerwins committed:
replace edge with ie13 and ie14

hackerwins committed:
replace ie13 and ie14 with latest.

hackerwins committed:
Merge pull request #2249 from summernote/misc/check-browers

uncomment tests with IE9 and Edge.

hackerwins committed:
Merge pull request #2212 from lacivert/develop

fix statusbar shown on disableResizeEditor

hackerwins committed:
Merge pull request #2180 from musaffa/fix-oncreatelink

[Fix] dont add default URL schema with onCreateLink

hackerwins committed:
Merge pull request #2197 from pvenakis/develop

Greek translation

hackerwins committed:
fixed indent error.

hackerwins committed:
Merge pull request #2185 from mstred/feature/ondestroy

Add support to an onDestroy event callback

hackerwins committed:
Merge pull request #2277 from summernote/develop

v0.8.3

hackerwins committed:
Update sit files

hackerwins committed:
add language files.

hackerwins committed:
Update dist files

Commit:
Better regex to match legit shorter video ids (ex.: https://vimeo.com/2)

Commit:
Better regex to match legit shorter video ids (ex.: https://vimeo.com/2)

hackerwins committed:
Merge pull request #2281 from ppython/videodialog-dailymotion-regex-fix

Better regex to match legit shorter vimeo video ids (ex.: https://vimeo.com/2)

geraldhumphries committed:
fix InvalidCharacterError when pasting an image in Firefox

fix #2283

lacivert committed:
typo for a Turkish word

correct word is "yaslamak" -> "yaslar". not "yazlar"

hackerwins committed:
Merge pull request #2285 from lacivert/patch-1

typo correction for a Turkish word

hackerwins committed:
fix karma version to 1.5.0 (tests were broken on 1.6.0)

hackerwins committed:
Merge pull request #2284 from geraldhumphries/bugfix/#2283

fix InvalidCharacterError when pasting an image in Firefox

hackerwins committed:
Replace `String.starsWith` with indexOf for compatibility for IE.
https://www.w3schools.com/jsref/jsref_startswith.asp

hackerwins committed:
Merge pull request #2178 from argelog/develop

Math Functions And Science Characters hint avaiable.(=)

Commit:
Rename wrong filename

Commit:
Add singleRun option into karma:all

hackerwins committed:
Merge pull request #2317 from summernote/minor/fix-karma-singlerun

Add singleRun option into karma:all

hackerwins committed:
Merge pull request #2316 from summernote/minor/fix-lang-mn

Rename wrong filename of summernote-mn-MN

hackerwins committed:
Merge pull request #2319 from summernote/master

Master to Develop

IonutVornicescu committed:
Correct exit from fullscreen

When exit from full screen on line #37 there where removed all windows resize callback function registerd to the event.
I changed the code in order to remove only the callback added by the plugin.

IonutVornicescu committed:
Update Fullscreen.js

jaguaryang committed:
修改打开新窗口和checkbox库混用bug

Commit:
Added target blank option

WaKeMaTTa committed:
(fr-FR) Add specialChar locales

:globe_with_meridians: (fr-FR) Add "specialChar" locales

hackerwins committed:
Merge pull request #2324 from WaKeMaTTa/patch-1

(locale fr-FR) Add specialChar

Commit:
Remove reference to CONTRIBUTING.md file

which has been removed

Commit:
Remove need to install global packages

as well as providing a facade for Tooling with NPM Package Scripts

Those are best practices described in the following articles:

* https://www.joezimjs.com/javascript/no-more-global-npm-packages/
* https://bocoup.com/blog/a-facade-for-tooling-with-npm-scripts/

And another article along the same lines regarding everything that can
be done with npm:
https://www.keithcirkel.co.uk/how-to-use-npm-as-a-build-tool/

Commit:
npm as a facade for server task and update README

hackerwins committed:
Merge pull request #2326 from madarche/fix-no-more-global-packages

Remove need to install global packages

hackerwins committed:
Merge pull request #2325 from madarche/fix-doc

Remove reference to CONTRIBUTING.md file

hackerwins committed:
Merge pull request #2323 from jaguaryang/develop

LinkDialog and some checkbox library mixed, checkbox state is not visible

hackerwins committed:
added prefix `sn-` for id attribute. related with #2323.

hackerwins committed:
Merge pull request #2320 from IonutVornicescu/patch-1

Correct exit from fullscreen

hackerwins committed:
Extract resizeTo method from toggle method. related with #2320.

hackerwins committed:
Prevent removing another event callback on window/document. related with #2320.

hackerwins committed:
Fixed wrong position of checkbox after merging #2323.

hackerwins committed:
Merge pull request #2195 from dmitriy-strukov/add-target-blank

Feature | Added target blank option

hackerwins committed:
Prevent overwriting default link target option with editing state. related with #2195.

hackerwins committed:
Merge pull request #2160 from czimnick/patch-1

fix for https://github.com/summernote/angular-summernote/issues/162

hackerwins committed:
added comments.

hackerwins committed:
Merge pull request #1917 from summernote/minor/disable-tooltip-by-option

Tooltip can be shown/hidden by settings

hackerwins committed:
Please don't include dist/* files on your commits.

hackerwins committed:
Merge pull request #2086 from latifz1/develop

508 compliance

hackerwins committed:
Merge branch 'develop' into bugfix/#1674

hackerwins committed:
Merge pull request #1919 from mstred/feature/custom_styletags

Add support to custom styletags right into the style toolbar menu

hackerwins committed:
Skip buttons.spec tests on firefox.

hackerwins committed:
Merge branch 'develop' of github.com:summernote/summernote into develop

hackerwins committed:
Skip buttons.spec tests on Edge.

reinaldocoelho committed:
Merge branch 'develop' into feature/table-popover-dist

# Conflicts:
#	dist/font/summernote.eot
#	dist/font/summernote.ttf
#	dist/font/summernote.woff
#	dist/lang/summernote-ar-AR.min.js
#	dist/lang/summernote-bg-BG.min.js
#	dist/lang/summernote-ca-ES.min.js
#	dist/lang/summernote-cs-CZ.min.js
#	dist/lang/summernote-da-DK.min.js
#	dist/lang/summernote-de-DE.min.js
#	dist/lang/summernote-es-ES.min.js
#	dist/lang/summernote-es-EU.min.js
#	dist/lang/summernote-fa-IR.min.js
#	dist/lang/summernote-fi-FI.min.js
#	dist/lang/summernote-fr-FR.js
#	dist/lang/summernote-fr-FR.min.js
#	dist/lang/summernote-gl-ES.min.js
#	dist/lang/summernote-he-IL.min.js
#	dist/lang/summernote-hr-HR.min.js
#	dist/lang/summernote-hu-HU.min.js
#	dist/lang/summernote-id-ID.min.js
#	dist/lang/summernote-it-IT.min.js
#	dist/lang/summernote-ja-JP.min.js
#	dist/lang/summernote-ko-KR.min.js
#	dist/lang/summernote-lt-LT.min.js
#	dist/lang/summernote-lt-LV.min.js
#	dist/lang/summernote-nb-NO.min.js
#	dist/lang/summernote-nl-NL.min.js
#	dist/lang/summernote-pl-PL.min.js
#	dist/lang/summernote-pt-BR.min.js
#	dist/lang/summernote-pt-PT.min.js
#	dist/lang/summernote-ro-RO.min.js
#	dist/lang/summernote-ru-RU.min.js
#	dist/lang/summernote-sk-SK.min.js
#	dist/lang/summernote-sl-SI.min.js
#	dist/lang/summernote-sr-RS-Latin.min.js
#	dist/lang/summernote-sr-RS.min.js
#	dist/lang/summernote-sv-SE.min.js
#	dist/lang/summernote-th-TH.min.js
#	dist/lang/summernote-tr-TR.js
#	dist/lang/summernote-tr-TR.min.js
#	dist/lang/summernote-uk-UA.min.js
#	dist/lang/summernote-vi-VN.min.js
#	dist/lang/summernote-zh-CN.min.js
#	dist/lang/summernote-zh-TW.min.js
#	dist/plugin/databasic/summernote-ext-databasic.min.js
#	dist/plugin/hello/summernote-ext-hello.min.js
#	dist/plugin/specialchars/summernote-ext-specialchars.min.js
#	dist/summernote.css
#	dist/summernote.js
#	dist/summernote.min.js

reinaldocoelho committed:
New dist created after merge and conflict solve with V0.8.3.

reinaldocoelho committed:
Removing dist files.

reinaldocoelho committed:
Adding unit tests to table process

reinaldocoelho committed:
Adding support to remove table column with colspan attribute

reinaldocoelho committed:
Adding support to remove table row with columns with rowspan attribute

reinaldocoelho committed:
Change table structure to process Col delete with colspan and rowspan.

reinaldocoelho committed:
change table work with rowspan and colspan for delete col and row.

reinaldocoelho committed:
Adding test case for delete row with virtual span cell.

reinaldocoelho committed:
change add test to insert row on top of rowspan

reinaldocoelho committed:
Solve bug when add line bellow rowspan

reinaldocoelho committed:
Add suport to add column when have rowspan in table

reinaldocoelho committed:
change to test remove column after colspan

reinaldocoelho committed:
change to test add column before colspan column
